### PR TITLE
Added deprecation notice

### PR DESCRIPTION
**Pull request by Ssofja**
# What does this PR do ?
Some deprecation notices have added for future code refactoring 
Add a one line overview of what this PR aims to accomplish.

**Collection**: [asr]
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [x] Documentation

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

### CODE REVIEW

File: nemo/collections/asr/data/audio_to_text.py [hunk 1]
```diff
@@ -730,6 +731,7 @@ def __call__(self, *args):
         )
 
 
+@deprecated(explanation='It will be removed in future versions, please use LhotseSpeechToTextBpeDataset class ')
```
Comment (nithinraok) [inline nemo/collections/asr/data/audio_to_text.py:734]:
```suggestion
@deprecated(explanation='Webdataset support will be removed in r2.1.0 versions, please use LhotseSpeechToTextBpeDataset class instead')
```

File: nemo/collections/asr/models/classification_models.py [hunk 2]
```diff
@@ -483,6 +484,7 @@ def get_transcribe_config(cls) -> ClassificationInferConfig:
         return ClassificationInferConfig()
 
 
+@deprecated(explanation='It will be replaced with EncDecSpeakerLabelModel class.')
```
Comment (nithinraok) [inline nemo/collections/asr/models/classification_models.py:487]:
```suggestion
@deprecated(explanation='EncDecClassificationModel will be merged with EncDecSpeakerLabelModel class.')
```

File: nemo/collections/asr/data/audio_to_text.py [hunk 3]
```diff
@@ -730,6 +731,9 @@ def __call__(self, *args):
         )
 
 
+@deprecated(
+    explanation='Webdataset support will be removed in r2.1.0 versions, please use LhotseSpeechToTextBpeDataset class instead'
+)
```
Comment (ko3n1g) [inline nemo/collections/asr/data/audio_to_text.py:736]:
```suggestion
    explanation='Webdataset support will be removed in v2.1.0 versions, please use LhotseSpeechToTextBpeDataset class instead'
)
```

**Review by ko3n1g — CHANGES_REQUESTED** [review]
small nit

File: nemo/collections/asr/models/confidence_ensemble.py [hunk 4]
```diff
@@ -151,6 +152,7 @@ def compute_confidence(hypothesis: Hypothesis, confidence_cfg: ConfidenceConfig)
     return conf_value
 
 
+@deprecated(version='2.1.0 NeMo')
```
Comment (titu1994) [inline nemo/collections/asr/models/confidence_ensemble.py:155]:
```suggestion
@deprecated(version='v2.1.0')
```

**Review by titu1994 — APPROVED** [review]
LGTM thanks !

**Review by nithinraok — APPROVED** [review]

**Pull request merged**